connect/transforms/src/main/java/org/apache/kafka/connect/transforms/field/SingleFieldPath.java

```diff
@@ -0,0 +1,240 @@
+                        steps.add(field.toString());
+                        idx++;
+                        break;
+                    }
+
+                    if (path.charAt(idx + 1) != DOT) { // not followed by a dot
+                        // this backtick isn't followed by a dot; include it in the field name, but continue
+                        // looking for a matching backtick that is followed by a dot
+                        idx++;
+                        continue;
+                    }
+
+                    if (escaped) {
+                        // this backtick was escaped; include it in the field name, but continue
+                        // looking for an unescaped matching backtick
+                        field.append(path, start, idx - 1)
+                            .append(BACKTICK);
+
+                        idx++;
+                        start = idx;
+                        continue;
+                    }
+
+                    // we've found our matching backtick
+                    field.append(path, start, idx);
+                    steps.add(field.toString());
+                    idx += 2; // increment by two to include the backtick and the dot after it
+                    break;
+                }
+            }
+        }
+        // add last step if last char is a dot
+        if (!path.isEmpty() && path.charAt(path.length() - 1) == DOT)
+            steps.add("");
+        return Collections.unmodifiableList(steps);
+    }
+
+    private static void failWhenIncompleteBacktickPair(String path, int backtickAt) {
+        throw new ConfigException("Incomplete backtick pair in path: [" + path + "],"
+                + " consider adding a backslash before backtick at position " + backtickAt
+                + " to escape it");
+    }
+
+    /**
+     * Access a {@code Field} at the current path within a schema {@code Schema}
+     * If field is not found, then {@code null} is returned.
+     */
+    public Field fieldFrom(Schema schema) {
+        if (schema == null) return null;
+
+        Schema current = schema;
+        for (String pathSegment : stepsWithoutLast()) {
+            final Field field = current.field(pathSegment);
+            if (field != null) {
+                current = field.schema();
+            } else {
+                return null;
+            }
+        }
+        return current.field(lastStep());
+    }
+
+    /**
+     * Access a value at the current path within a schema-based {@code Struct}
+     * If object is not found, then {@code null} is returned.
+     */
+    public Object valueFrom(Struct struct) {
+        if (struct == null) return null;
+
+        Struct current = struct;
+        for (String pathSegment : stepsWithoutLast()) {
+            // Check to see if the field actually exists
+            if (current.schema().field(pathSegment) == null) {
+                return null;
+            }
+            current = current.getStruct(pathSegment);
```

Comment:
Nit: we could have slightly friendlier error messages here:
```suggestion
            Object subValue = current.get(pathSegment);
            current = requireStructOrNull(subValue, "nested field access");
```